### PR TITLE
add : golang code sample for detecting faces in images

### PR DESCRIPTION
**Pull request by 3l-d1abl0**
Issue https://github.com/awsdocs/amazon-rekognition-developer-guide/issues/35, 

*Description of changes:*

Added Golang code sample for detecting faces in an Image.
Let me know if any further changes are needed .

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: doc_source/faces-detect-images.md [hunk 1]
```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
       });
    ```
 
+------
+#### [ GO ]
+
+   This example displays the estimated age range , position , emotion and Gender for each detected faces\. Change the value of `photo` to the image file name\. Change the value of `bucket` to the Amazon S3 bucket where the image is stored\.
+
+   ```
+   //Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (Doug-AWS) [inline doc_source/faces-detect-images.md:359]: All of our code examples use the following copyright:

// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX - License - Identifier: Apache - 2.0

File: doc_source/faces-detect-images.md [hunk 2]
```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
...
+   //PDX-License-Identifier: MIT-0 (For details, see https://github.com/awsdocs/amazon-rekognition-developer-guide/blob/master/LICENSE-SAMPLECODE.)
+
+   package main
+
+    import (
+        "fmt"
```
Comment (Doug-AWS) [inline doc_source/faces-detect-images.md:365]: Add "flag" (see main).

File: doc_source/faces-detect-images.md [hunk 3]
```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
...
+            fmt.Println("Error while creating session,", err)
+            return
+        }
+
+        svc = rekognition.New(sess)
+        _ = svc
+    }
+
+    func main() {
+
```
Comment (Doug-AWS) [inline doc_source/faces-detect-images.md:393]: We're moving away from user's having to modify the code and using command-line flags:

bucket := flag.String("b", "", "The name of the bucket")
photo := flag.String("p", "", "The path to the photo file (JPEG, GIF, or ???")
flag.Parse()

if *bucket == "" || *photo == "" {
	fmt.Println("You must supply a bucket name (-b BUCKET) and photo file (-p PHOTO")
	return
}

File: doc_source/faces-detect-images.md [hunk 4]
```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
...
+        bucket := "" //Bucket name from S3
+        photo := ""  //Image File
+
+        params := &rekognition.DetectFacesInput{
+            Image: &rekognition.Image{ // Required
+
+                S3Object: &rekognition.S3Object{
+                    Bucket: aws.String(bucket),
```
Comment (Doug-AWS) [inline doc_source/faces-detect-images.md:401]: If you use the earlier flags, this becomes:

Bucket: bucket,
Name: photo,

File: doc_source/faces-detect-images.md [hunk 5]
```diff
@@ -350,6 +350,92 @@ You can provide the input image as an image byte array \(base64\-encoded image b
...
+                },
+            },
+            Attributes: []*string{
+                aws.String("ALL"), // Required
+            },
+        }
+
+        resp, err := svc.DetectFaces(params)
+
+        if err != nil { //If there is an Error
```
Comment (Doug-AWS) [inline doc_source/faces-detect-images.md:412]: Delete the comment. Go style is no space between DetectFaces call and error test.

**Review by Doug-AWS — CHANGES_REQUESTED** [review]

File: doc_source/faces-detect-images.md (hunk 1 above)
Comment (3l-d1abl0) [reply doc_source/faces-detect-images.md]: Hi Doug,
I have updated the copyright statement.

File: doc_source/faces-detect-images.md (hunk 2 above)
Comment (3l-d1abl0) [reply doc_source/faces-detect-images.md:366]: Addressed this as well.

File: doc_source/faces-detect-images.md (hunk 3 above)
Comment (3l-d1abl0) [reply doc_source/faces-detect-images.md:405]: Done. I have added the command-line parsing to init()

File: doc_source/faces-detect-images.md (hunk 4 above)
Comment (3l-d1abl0) [reply doc_source/faces-detect-images.md:410]: Addressed.

File: doc_source/faces-detect-images.md (hunk 5 above)
Comment (3l-d1abl0) [reply doc_source/faces-detect-images.md]: Adressed !
Let me know  if theses changes are fine. I will change my other PRs as well.

**Comment by 3l-d1abl0** [comment]
HI @Doug-AWS 
Did you get a chance to review this ?

**Comment by Doug-AWS** [comment]
We are actively discouraging embedding code examples in our service guides, such as this one for Amazon Rekognition. Instead, we create them in our [aws-doc-sdk-examples](https://github.com/awsdocs/aws-doc-sdk-examples) repo in a directory/sub-directory based on SDK and service, include the code example in an SDK developer guide, and add a link to the SDK dev guide code example topic to the service guide.

Since your submission includes code examples for Java, the CLI, Python, .NET (C#), Ruby, Node.js (JavaScript) and Go, you will need to submit this as 7 distinct PRs in our GitHub repo. Note that there are already Amazon Rekognition code examples in [Java](https://github.com/awsdocs/aws-doc-sdk-examples/tree/master/javav2/example_code/rekognition), [Python](https://github.com/awsdocs/aws-doc-sdk-examples/tree/master/python/example_code/rekognition), and [JavaScript](https://github.com/awsdocs/aws-doc-sdk-examples/tree/master/javascriptv3/example_code/rekognition/estimate-age-example/src), so be sure your code example differs from the existing code examples.

Once your PRs are submitted, one of our team will take a look at each one and provide any feedback.

**Comment by 3l-d1abl0** [comment]
Thank you @Doug-AWS !
I will surely take a look at that repo and raise PR's accordingly.

**Pull request closed without merging**